pkg/nfd-master/nfd-api-controller.go

```diff
@@ -41,19 +41,23 @@ type nfdController struct {
 	updateOneNodeChan  chan string
 }
 
-func newNfdController(config *restclient.Config, disableNodeFeature bool) (*nfdController, error) {
+type NfdApiControllerOptions struct {
```

Comment:
nit: we don't need to export this (especially because `newNfdController()` is not exported:

```suggestion
type nfdApiControllerOptions struct {
```